Added Missing use_federated_token Parameter to Azure Storage Configuration Docs for Workload Identity (#3236)
grafana/tempo: docs/sources/tempo/configuration/azure.md

Tempo requires the following configuration to authenticate to and access Azure blob storage:

- Storage Account name specified in the configuration file as `storage_account_name` or in the environment variable `AZURE_STORAGE_ACCOUNT`

> - Credentials for accessing the Storage Account; can be one of the following:

Reviewer: Changing ; to a period to match the other bullets at the same level. 
```suggestion
- Credentials for accessing the Storage Account.  Can be one of the following:
```

> - Credentials for accessing the Storage Account; can be one of the following:
>   - Storage Account access key specified in the configuration file as `storage_account_key` or in the environment variable `AZURE_STORAGE_KEY`
>   - An Azure Managed Identity; either system or user assigned. To use Azure Managed Identities, you'll need to set `use_managed_identity` to `true` in the configuration file or set `user_assigned_id` to the client ID for the managed identity you'd like to use.
>       - For a system-assigned managed identity, no additional configuration is required.
>       - For a user-assigned managed identity, you'll need to set `user_assigned_id` to the client ID for the managed identity in the configuration file.

Reviewer: Updating to have it read a little better. 
```suggestion
- Credentials for accessing the Storage Account that are one of the following:
  - Storage Account access key specified in the configuration file as `storage_account_key` or in the environment variable `AZURE_STORAGE_KEY`
  - An Azure Managed Identity that is either system or user assigned. To use Azure Managed Identities, you'll need to set `use_managed_identity` to `true` in the configuration file or set `user_assigned_id` to the client ID for the managed identity you'd like to use.
      - For a system-assigned managed identity, no additional configuration is required.
      - For a user-assigned managed identity, you'll need to set `user_assigned_id` to the client ID for the managed identity in the configuration file.
```

  - Via Azure Workload Identity. To use Azure Workload Identity, you'll need to enable Azure Workload Identity on your cluster, add the required label and annotation to the service account and the required pod label. Additionally, you will need to set `use_federated_token` to `true` to utilize Azure Workload Identity.

> ## Sample configuration (for Tempo Monolithic Mode)

Reviewer: This is one of the headings that was modified by the PR conflict resolution. Is your example okay in this section?

Author: LGTM :smile:
